prov/tcp/src/tcpx.h

```diff
@@ -187,13 +188,19 @@ struct tcpx_pep {
 	struct poll_fd_info	poll_info;
 };
 
+struct tcpx_cm_state {
+	fastlock_t		lock;
+	int			shutdown_reported;
+};
+
 struct tcpx_ep {
 	struct util_ep		util_ep;
 	SOCKET			conn_fd;
 	struct dlist_entry	rx_queue;
 	struct dlist_entry	tx_queue;
 	/* lock for accessing rx & tx queues */
 	fastlock_t		queue_lock;
+	struct tcpx_cm_state	cm_state;
```

Comment:
Just embed the 2 fields directly into this structure.  If a state is needed, a generic state or set of flags would be preferable than a simply boolean to track whether shutdown has been reported.